teemaderegister-react/src/actions/CsvActions.js

```diff
@@ -0,0 +1,57 @@
+import { CSV_REQUEST } from '../constants/ApiConstants'
+import axios from 'axios'
+import moment from 'moment'
+import xlsx from 'xlsx'
+import { saveAs } from 'file-saver'
+const EXCEL_TYPE = 'application/csv'
+
+export const getCsvData = (status, course) => {
+  return axios.get(CSV_REQUEST, {params: {status: status, course: course}})
+    .then((res) => {
+      createCsv(res.data)
+    })
+    .catch((err) => {
+      console.log(err)
+    })
+}
+
+const createCsv = data => {
+  console.log(data[0].title)
+  let fileName = data[0].curriculums[0].abbreviation
+  let mappedData
+
+  if (typeof data[0].defended === 'undefined' && typeof data[0].registered !== 'undefined') {
```

Comment:
instead of if else block here make a function that only depending on input formats the data, e.g
```
formatrow = row => {
cons response = {
   Title: title
}

if (defended) {
response.defended = ...
...
}

data.map(formatRow)

```

Reply:
should be less lines of code